src/game/AI/ScriptDevAI/scripts/outland/tempest_keep/the_eye/boss_alar.cpp

```diff
@@ -144,6 +145,12 @@ struct boss_alarAI : public ScriptedAI
     {
         if (m_pInstance)
             m_pInstance->SetData(TYPE_ALAR, DONE);
+            
+        std::list<Player*> playerList;
+            GetPlayerListWithEntryInWorld(playerList, m_creature, 150.0f);
```

Comment:
I don't think you need to loop through the entire list of players.
The ones which are inside the instance, is enough.

Reply:
Hello, Radu.
1) Pull redone. https://github.com/cmangos/mangos-wotlk/pull/241
2) I don't know, how make otherwise. I'm not such an advanced scripting writer(
For quest credit need quest and player should have aura for quest complete.